content/js3/blocks/simplifying-element-creation/index.md

```diff
@@ -0,0 +1,166 @@
+
+```js
+function createChildElement(parentElement, tagName, textContent) {
+    const element = document.createElement(tagName);
+    element.textContent = textContent;
+    parentElement.append(element);
+    return element;
+}
+```
+
+And then rewrite our code to create the card like this:
+
+```js
+const film = {
+  title: "Killing of Flower Moon",
+  director: "Martin Scorsese",
+  times: ["15:35"],
+  certificate: "15",
+  duration: 112,
+};
+
+function createChildElement(parentElement, tagName, textContent) {
+    const element = document.createElement(tagName);
+    element.textContent = textContent;
+    parentElement.append(element);
+    return element;
+}
+
+const card = document.createElement("section");
+
+createChildElement(card, "h3", film.title);
+
+createChildElement(card, "p", `Director: ${film.director}`);
+
+const times = createChildElement(card, "ul", "");
+
+for (const time of film.times) {
+  createChildElement(times, "li", time);
+}
+
+createChildElement(card, "p", `${film.duration} minutes`);
+
+createChildElement(card, "p", `Certificate: ${film.certificate}`);
+
+document.body.append(card);
+```
+
+This code does exactly the same thing as the code we had before. By introducing a function we have introduced some advantages:
+1. Our code is smaller, which can make it easier to read and understand what it's doing.
+2. If we want to change how we create elements (e.g. if we wanted to also add a class attribute for each element), we only need to write the new code one time, not for every element.
+3. We can see that each of the elements is being created the same way. Before, if we wanted to make sure the director element and duration element were created the same, we would have to compare several lines of code. Because we can see they're calling the same function, we know they're made the same way.
+
+There are also some drawbacks to our refactoring:
+1. If we want to change how we create some elements (e.g. we want to include an image of the director, replace the certificate text with a symbol, etc), we may have made it _harder_ to make these changes.
+2. If we want to see how one particlar thing, like the director, is show, we need to look in a few places. This is something you will need to get used to, so it's good to start practising now!
+
+### Using `<template>` tags
+
+A different way we could simplify this code is to use a different technique for creating elements.
+
+Until now, we have only seen one way to create elements: `document.createElement`.
+
+The DOM has another way of creating elements - we can _copy_ existing elements and then change them.
+
+HTML has a useful tag designed to help make this easy, [the `<template>` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template). When you add a `<template>` element to a page, it doesn't get displayed when the page loads.
+
+We can copy any DOM element, not just `<template>` tags. For this problem, we will uus a `<template>` tag.
```

Comment:
```suggestion
We can copy any DOM element, not just `<template>` tags. For this problem, we will use a `<template>` tag because it has some special powers.
```